errors.go

```diff
@@ -151,6 +152,24 @@ func GetUserMessage(err error) string {
 	return ""
 }
 
+// errorDetailer identifies an error with details
+type errorDetailer interface {
+	Details() []interface{}
+}
+
+// Details returns the error details
+func (c *customError) Details() []interface{} { return c.details }
+
+// GetDetails returns a slice of arbitratry details for all errors
+// If error is not `errorDetailer` returns empty slice
+func GetDetails(err error) []interface{} {
+	if detailedError, ok := err.(errorDetailer); ok {
+		return detailedError.Details()
+	}
+
+	return []interface{}{}
```

Comment:
returning `nil` is cleaner and has the same impact when you append to it
see https://play.golang.org/p/UVO9hyzVcpn

Reply:
It works! Didn't realize, but makes sense. Updated it.